manticore/platforms/evm.py

```diff
@@ -2566,102 +2567,63 @@ def set_storage_data(self, storage_address, offset, value):
         :param value: the value to write
         :type value: int or BitVec
         """
-        self._world_state[storage_address]["storage"][offset] = value
-
-    def get_storage_items(self, address):
-        """
-        Gets all items in an account storage
+        self._world_state.set_storage_data(self.constraints, storage_address, offset, value)
 
-        :param address: account address
-        :return: all items in account storage. items are tuple of (index, value). value can be symbolic
-        :rtype: list[(storage_index, storage_value)]
-        """
-        storage = self._world_state[address]["storage"]
-        items = []
-        array = storage.array
-        while not isinstance(array, ArrayVariable):
-            items.append((array.index, array.value))
-            array = array.array
-        return items
-
-    def has_storage(self, address):
+    def has_storage(self, address: int) -> bool:
         """
         True if something has been written to the storage.
         Note that if a slot has been erased from the storage this function may
         lose any meaning.
         """
-        storage = self._world_state[address]["storage"]
-        array = storage.array
-        while not isinstance(array, ArrayVariable):
-            if isinstance(array, ArrayStore):
-                return True
-            array = array.array
-        return False
-
-    def get_storage(self, address):
```

Comment:
The concern is that the lack of this method could break external scripts?

I ask because the old `get_storage` method returned a `Dict` or `ArrayProxy` (I think). I am afraid that supporting that API faithfully could be a heavy lift.

Reply:
> The concern is that the lack of this method could break external scripts?

Yeah, since the old method was reasonably well-documented and appeared on a fairly prominent object, I'm worried that removing it might break user scripts. Then again, I'm not entirely sure what the purpose of this function is. It looks like it was called in a few places internally, which you've replaced with calls to `_get_storage`. 

Ideally we'd have a deprecation period before removing any APIs, but this change should be okay so long as we document it, and there's some kind of workaround available to achieve the same functionality as we had before. Would it make sense to have `get_storage` as a this wrapper around `_get_storage` and instruct users that they'll need to process the returned data somewhat differently? Or should we direct them to something like `get_storage_data` or `get_storage_items`?

Reply:
I re-added the method with return type `Storage`:
https://github.com/trailofbits/manticore/blob/954e92f08067745f8d5a004bc97acc770517ddad/manticore/platforms/evm.py#L2580
And I implemented `__getitem__` for `Storage`:
https://github.com/trailofbits/manticore/blob/954e92f08067745f8d5a004bc97acc770517ddad/manticore/platforms/evm_world_state.py#L48
So, this should satisfy any users of this method, regardless of whether they were expecting a dictionary or symbolic array.